### PR TITLE
Make RenderUiKitView reject absorbed touch events

### PR DESCRIPTION
**Pull request by amirh**
## Description

When a touch event that is in the bounds of a RenderUiKitView is absorbed by another render object,
the RenderUiKitView's handleEvent is not called for that object. On the platform side, the touch event hits the FlutterTouchInterceptingView which is waiting for a framework decision that never arrived on whether to reject or accept the gesture.

This change fixes the issue by having RenderUiKitView register a global PointerRoute, that is used to reject absorbed touch events.

## Related Issues

https://github.com/flutter/flutter/issues/28659

### CODE REVIEW

File: packages/flutter/lib/src/rendering/platform_view.dart [hunk 1]
```diff
@@ -349,13 +351,41 @@ class RenderUiKitView extends RenderBox {
 
   @override
   void handleEvent(PointerEvent event, HitTestEntry entry) {
-    if (event is PointerDownEvent) {
-      _gestureRecognizer.addPointer(event);
+    if (!(event is PointerDownEvent)) {
```
Comment (dnfield) [inline packages/flutter/lib/src/rendering/platform_view.dart]: nit: `if (event is! PointerDownEvent) {`

Comment (amirh) [reply packages/flutter/lib/src/rendering/platform_view.dart]: done

**Comment by dnfield** [comment]
This LGTM, though I suspect it would be good to get a review from someone else more familiar with the gesture system.

**Review by cyanglaz — APPROVED** [review]
download

**Pull request merged**